src/os/bluestore/BlueStore.cc

```diff
@@ -6859,7 +6861,12 @@ int BlueStore::mkfs()
   dout(10) << " freelist_type " << freelist_type << dendl;
 
   // choose min_alloc_size
-  if (cct->_conf->bluestore_min_alloc_size) {
+  if ((cct->_conf->bluestore_use_optimal_io_size_for_min_alloc_size) && (optimal_io_size != 0)) {
```

Comment:
It might be helpful to print optimal_io_size unconditionally to be able to investigate what's the reported value is.

Reply:
Additionally I'd like to have a sort of protection - assertion or falling back to regular min_alloc_size when optimal_io_size is below/not aligned with block_size...

Reply:
Hi @ifed01 - good points.  I've added both of your suggestions/requests.   Thanks.